core/src/smartcontracts/isi/domain.rs

```diff
@@ -85,6 +83,15 @@ pub mod isi {
                 .map_err(Error::Validate)?;
 
             let asset_definition_id = asset_definition.id().clone();
+            let latest_tx_hash = wsv
+                .latest_transaction_hash()
```

Comment:
So you still use latest hash, but not the current hash. Your test shouldn't pass then